backend/src/main/kotlin/uk/gov/justice/digital/backend/client/preview/AthenaPreviewClient.kt

```diff
@@ -0,0 +1,39 @@
+package uk.gov.justice.digital.backend.client.preview
+
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+import org.slf4j.LoggerFactory
+import javax.sql.DataSource
+
+@Singleton
+class AthenaPreviewClient(@Named("preview") private val previewDataSource: DataSource) : PreviewClient {
+
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    override fun runQuery(query: String): List<Map<String, String>> {
+        logger.info("Executing query: {}", query)
+        val startTime = System.currentTimeMillis()
+        val statement = previewDataSource.connection.createStatement()
+        val resultSet = statement.executeQuery(query)
+        val duration = System.currentTimeMillis() - startTime
+        logger.info("Query executed successfully in {}ms", duration)
+
+        val metadata = resultSet.metaData
+        val columnCount = metadata.columnCount
+
+        // Column names start from 1
+        val columnNames = (1..columnCount)
+            .map { metadata.getColumnName(it) }
+
+        val result = mutableListOf<Map<String, String>>()
+
+        while(resultSet.next()) {
```

Comment:
Minor: Is there any limit of the length of this data in the query (or in the dataset)? This could be quite a large amount of stuff in memory.

Reply:
Good point. The maximum number of rows is currently set at 100 which is something of a guess. This can be decreased if it turns out to be too big. I'll be keeping an eye on the mem usage metrics as I run further tests. So far I've not seen anything worrying.